cmd/metacache-entries.go

```diff
@@ -89,6 +89,38 @@ func (e *metaCacheEntry) matches(other *metaCacheEntry, bucket string) bool {
 	return eFi.ModTime.Equal(oFi.ModTime) && eFi.Size == oFi.Size && eFi.VersionID == oFi.VersionID
 }
 
+// resolveEntries returns if the entries match by comparing their latest version fileinfo.
+func resolveEntries(a, b *metaCacheEntry, bucket string) *metaCacheEntry {
+	if b == nil {
+		return a
+	}
+	if a == nil {
+		return b
+	}
+
+	aFi, err := a.fileInfo(bucket)
```

Comment:
Should we not get fileInfo for both and then decide to return the correct entry based on that conversion?

Reply:
If decoding `a` fails, we return `b` and vice versa.

There is no guarantee that the returned value is useful, we just need to choose which one we'd prefer to keep.